cloud/azure/deploy/website.go

```diff
@@ -518,3 +690,30 @@ func (p *NitricAzurePulumiProvider) deployCDN(ctx *pulumi.Context) error {
 
 	return nil
 }
+
+func getPrimaryEndpointUrl(storageAccount *storage.StorageAccount) pulumi.StringOutput {
+	return storageAccount.PrimaryEndpoints.ApplyT(func(endpoints storage.EndpointsResponse) (string, error) {
+		parsed, err := url.Parse(endpoints.Web)
+		if err != nil {
+			return "", err
+		}
+		return parsed.Hostname(), nil
+	}).(pulumi.StringOutput)
+}
+
+// Convert a name to a unique number
+// This creates a number that's guaranteed unique for different strings
+func nameToUniqueNumber(name string) int {
+	// Start at a high base to avoid conflicts with other rules
+	base := 10000
+
+	// Use character position and value to guarantee uniqueness
+	// This is essentially creating a custom numeric representation
+	for i, char := range name {
+		// Multiply by position+1 to weight characters differently
+		// Use prime number multiplier to reduce collision risk
+		base += int(char) * (i + 1) * 31
+	}
+
+	return base
+}
```

Comment:
If this method is here to workaround issues with rule order conflicts for provisioning new rules before old rules are deleted make sure its clearly documented, without clear documentation is looks like it shouldn't be here.